### PR TITLE
added basic BeanPostProcessor

### PR DESCRIPTION
**Pull request by NikolayProkopchuk**
Added basic logic for BeanPostprocessors. There are no concrete postprocessors yet. After bean creation will be called postProcessBeforeInitialization method for this bean in each postprocessor found in package

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -35,7 +36,10 @@ repositories {
 dependencies {
     implementation "org.reflections:reflections:${reflections}"
     implementation "org.projectlombok:lombok:${lombok}"
+    annotationProcessor "org.projectlombok:lombok:${lombok}"
```
Comment (EEM86) [inline build.gradle:39]: Doesn't implementation handle it?

File: src/main/java/com/bobocode/svydovets/beans/factory/DefaultListableBeanFactory.java [hunk 2]
```diff
@@ -13,11 +14,23 @@
...
+
+@Log4j2
 public class DefaultListableBeanFactory implements BeanFactory {
 
+    @Getter
+    private final Set<BeanPostProcessor> beanPostProcessors;
+
+    public DefaultListableBeanFactory(Set<BeanPostProcessor> beanPostProcessors) {
+        this.beanPostProcessors = beanPostProcessors;
+    }
```
Comment (EEM86) [inline src/main/java/com/bobocode/svydovets/beans/factory/DefaultListableBeanFactory.java:32]: Let's use @RequiredArgsConstructor?

File: src/main/java/com/bobocode/svydovets/beans/scanner/BeanPostprocessorScanner.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+     * This method scan package to find classes implemented
+     * {@link BeanPostProcessor} and returns set of bean postprocessors
+     * created from these classes.
+     *
+     * @param packageName - package name that will be scanned
+     * @return - map of bean definition
+     */
+    public Set<BeanPostProcessor> scan(String packageName) {
+        Objects.requireNonNull(packageName, "The packageName cannot be null! Please specify the packageName.");
+        if (StringUtils.isEmpty(packageName)) {
```
Comment (EEM86) [inline src/main/java/com/bobocode/svydovets/beans/scanner/BeanPostprocessorScanner.java:26]: As an option we can remove 25 lane and check for null or empty using StringUtils, change the message for it if you want to apply

**Review by EEM86 — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (NikolayProkopchuk) [reply build.gradle:39]: I'm not an expert in gradle, but when I remove this, the project is not built

**Review by khshanovskyi — APPROVED** [review]

**Pull request merged**